### ISSUE TITLE
MEDIUM: Implement preflight training validation

### ISSUE THREAD

**Issue by CryptoYogiLLC** [CryptoYogiLLC/vazhi#33]
## Problem
Previous training runs wasted days because problems weren't detected early:
- v3.1 mixed data formats produced "systemsystemsystem..." garbage
- v0.5 low loss (0.53) but garbage output
- v0.7 tokenizer corruption only discovered after full training

## Lesson Learned
Need fail-fast validation before committing to full training.

## Solution - Preflight System

```python
def preflight_validation(model, tokenizer, train_data, target_format='gguf'):
    """Run quick validation before full training."""
    print("🔍 Starting preflight validation...")
    
    # Step 1: Train on small sample (50 samples, 10 steps)
    print("  Step 1/5: Micro-training...")
    micro_train(model, train_data[:50], steps=10)
    
    # Step 2: Convert to target format
    print("  Step 2/5: Converting to GGUF...")
    gguf_path = convert_to_gguf(model, quantization='Q4_K_M')
    
    # Step 3: Test Tamil output quality
    print("  Step 3/5: Testing Tamil output...")
    test_prompts = [
        "வணக்கம், எப்படி இருக்கிறீர்கள்?",
        "திருக்குறள் முதல் குறள் என்ன?",
        "தமிழ்நாட்டின் தலைநகரம் என்ன?",
    ]
    
    for prompt in test_prompts:
        response = generate(gguf_path, prompt)
        tamil_pct = calculate_tamil_percentage(response)
        
        if tamil_pct < 30:
            raise PreflightError(f"Tamil output too low: {tamil_pct}%")
        if 'system' in response.lower() or '<think>' in response:
            raise PreflightError(f"Garbage output detected: {response[:100]}")
    
    # Step 4: Validate response coherence
    print("  Step 4/5: Checking coherence...")
    if len(response) < 10:
        raise PreflightError("Response too short")
    
    # Step 5: Memory/performance check
    print("  Step 5/5: Performance check...")
    
    print("✅ Preflight passed! Safe to proceed with full training.")
    return True
```

## Integration Point
Add to training notebooks before main training loop.

## Effort
4 hours

## References
- Consensus Report: docs/CODE_REVIEW_CONSENSUS_REPORT.md
- Expert: AI/ML
- Training Log: models/TRAINING_LOG.md (28 lessons learned)

**Comment by CryptoYogiLLC** [comment]
## Completed in commit 4c50403

### Changes Made:
Created `scripts/preflight_validation.py` with 5-step validation:

1. **Training data validation**:
   - Schema compliance check
   - Garbage pattern detection (system tokens, thinking tags, etc.)
   - Tamil content percentage analysis

2. **Model output testing** (if llama.cpp available):
   - First token latency check
   - Tamil output percentage validation
   - Coherence verification

3. **Tokenizer validation**:
   - Tamil vocabulary count check (requires 100+ Tamil tokens)

Usage:
```bash
python scripts/preflight_validation.py --data training.json --model model.gguf
```

Fail-fast validation prevents wasted training runs.
